src/test/java/com/esewa/usermanagement/service/AdminServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package com.esewa.usermanagement.service;
+
+import com.esewa.usermanagement.entity.RegistrationLog;
+import com.esewa.usermanagement.entity.User;
+import com.esewa.usermanagement.testobjectbuilder.TestUserBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class AdminServiceTest {
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private EmailService emailService;
+
+    @Mock
+    private RegistrationLogService registrationLogService;
+    @InjectMocks
+    private AdminServiceImpl adminService;
+
+    @Test
+    void givenValidUser_whenRegisterUser_thenSucceed() {
+
+        User user = TestUserBuilder.createUser();
+
+        when(userService.registerUser(user)).then(returnsFirstArg());
+        doNothing().when(emailService).sendEmail(user);
+        User actualResult = adminService.registerUser(user);
+
+        assertEquals(user, actualResult);
+    }
+
+    @Test
+    void givenValidUsers_whenRegisterMultipleUsers_thenSucceed() {
+
+        List<User> usersToRegister = TestUserBuilder.createUsers();
+
+        doNothing().when(emailService).sendEmail(any(User.class));
+        doNothing().when(registrationLogService).saveLog(any(RegistrationLog.class));
+        when(userService.registerUserAsync(any(User.class)))
+                .thenAnswer(invocation -> CompletableFuture.completedFuture(invocation.getArgument(0)));
+
+        List<User> registeredUsers = adminService.registerMultipleUsers(usersToRegister);
+
+        assertEquals(usersToRegister.size(), registeredUsers.size());
+        verify(userService, times(usersToRegister.size())).registerUserAsync(any(User.class));
+
+    }
+
+    @Test
+    void testRegisterMultipleUsers_throwsException_forBlacklistedUser() {
+
+        List<User> users = TestUserBuilder.createUsers();
+
+        doNothing().when(emailService).sendEmail(any(User.class));
+        doNothing().when(registrationLogService).saveLog(any(RegistrationLog.class));
+        when(userService.registerUserAsync(any(User.class)))
+                .thenAnswer(invocation -> {
+                    User user = invocation.getArgument(0);
+                    return !"shivam".equals(user.getName())
```

Comment:
For constants like "shivam", can you put this as a field of this class?

`public static final String BLACKLISTED_USER = "shivam";`

Reply:
Done